sch_cake.c

```diff
@@ -1110,7 +1123,7 @@ static struct sk_buff *cake_ack_filter(struct cake_sched_data *q,
 		 * sequence number or because it is a SACK
 		 */
 		if ((ntohl(tcph_check->ack_seq) == ntohl(tcph->ack_seq) &&
-		     !cake_tcph_is_sack(tcph)) ||
+			 (cake_tcph_sack_seq(tcph_check) >= cake_tcph_sack_seq(tcph))) ||
```

Comment:
You'll need to also increase the buffer size of the tcph_check variable, otherwise this can overflow

Reply:
Good catch, will fix.